pkg/kubelet/prober/prober_manager.go

```diff
@@ -253,8 +258,14 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *v1.PodStatus) {
 				ready = result == results.Success
 			} else {
 				// The check whether there is a probe which hasn't run yet.
-				_, exists := m.getWorker(podUID, c.Name, readiness)
-				ready = !exists
+				w, exists := m.getWorker(podUID, c.Name, readiness)
+				if exists {
+					// Trigger an immediate run of the readinessProbe to update ready state
+					w.doProbe()
```

Comment:
so this will run the probe synchronously? Will all the callers be OK with the long executing call?

Reply:
For that I need more thinking...

Reply:
Since this is an API call, I cannot control all calls to this one... however the `doProbe` call only happens in one particular case which is:
- container has started
- container has a readinessProbe
- readinessProbe has never been executed

But that's true the call would be blocking... avoiding it would require some sort of async defer or goroutine call? I am not too familiar with these, and we might end up with resource leaks or panics?

Reply:
Or maybe we need a special ticker which allows immediate calls (such as Reset but with immediate fire).